src/reconcile/rec_write.c

```diff
@@ -1994,16 +2017,19 @@ __rec_split(WT_SESSION_IMPL *session, WT_RECONCILE *r, size_t next_len)
 		/* If the next object fits into this page, we're good to go. */
 		if (r->space_avail >= next_len)
 			return (0);
+		break;
```

Comment:
@agorrod, this isn't right: if `space_avail < next_len`, we're going to break out of the case and grow the buffer, which isn't correct, we may simply be at the end of the page with a handful of bytes left and the next key/value pair doesn't fit.